types.go

```diff
@@ -533,7 +533,7 @@ func (r *refDynValue) getValue(
 		}
 		return nil, err
 	}
-	return parseValue(p, opts, str)
+	return parseValue(p, opts, str, parse.EnvParserConfig)
```

Comment:
this will affect parsing for every single resolver. `go-ucfg` has 3 resolver options by default: `ResolveEnv`, `ResolveNOOP`, and `Resolve`. The last one accepts a custom function.

I think it makes sense to have the resolver also configure the parsing mode.

Reply:
There is also a Keystore resolver in libbeat.